lib/cog_api/http/base.ex

```diff
@@ -52,7 +52,8 @@ defmodule CogApi.HTTP.Base do
 
   def find_id_by(%Endpoint{}=endpoint, resource, find_fun)
       when is_function(find_fun) do
-    with {:ok, %{^resource => items}} <- get(endpoint, resource) do
```

Comment:
Roles is using this when executing #show. https://github.com/operable/cog-api-client/blob/lhaskins/new-perm-in-roles/lib/cog_api/http/roles.ex#L13

This allows us to continue to get role information based on the role name when we do not have the id.